lib/c_glib/src/thrift/c_glib/protocol/thrift_protocol.c

```diff
@@ -61,15 +61,15 @@ thrift_protocol_set_property (GObject *object, guint property_id,
   switch (property_id)
   {
     case PROP_THRIFT_PROTOCOL_TRANSPORT:
-      protocol->transport = g_value_get_object (value);
```

Comment:
Ok. As long as the reference count is guaranteed to be well managed. And it seems because you added the finalize method it should be okay. But I think this breaks RAII. Since the application can get the resource and free it but the resource will outlive the requester inside this class. But I come from C/C++ and maybe RAII doesn't apply quite well here because the reference counter. I will study about it a little bit.